File: frontend/src/types/Recipe.ts
export interface Recipe {
  id: number
  name: string
  description: string | null
  ingredients: string | null
  cookTimeMinutes: number | null
  createdAt: string
  updatedAt: string
}

export interface RecipeUpdateRequest {
  name: string
  description?: string | null
  ingredients?: string | null
  cookTimeMinutes?: number | null
}

export interface RecipeCreateRequest {
  name: string
  description?: string | null
  ingredients?: string | null
  cookTimeMinutes?: number | null
}

export interface RecipeImageExtractResult {
  success: boolean
  name?: string | null
  description?: string | null
  ingredients?: string | null
  cookTimeMinutes?: number | null
  message?: string | null
}

File: backend/src/WhatsForDinner.Api/Models/Recipe.cs
namespace WhatsForDinner.Api.Models;

public class Recipe
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Ingredients { get; set; }
    public int? CookTimeMinutes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User User { get; set; } = null!;
    public ICollection<WeeklyPlanItem> WeeklyPlanItems { get; set; } = new List<WeeklyPlanItem>();
}
